### ISSUE TITLE
Add the option to modify the value of `epsilon` when using `MeanAbsolutePercentageError`

### ISSUE THREAD

**Issue by Abdelrhman-Hosny** [Lightning-AI/metrics#1230]
## 🚀 Feature

I'm proposing that we add the option to modify the value of epsilon when using the class version of MAPE
$\text{MAPE} = \frac{\hat y - y}{\max (y ,\epsilon)}$ 
### Motivation
I was using the library and found that there is no way to modify the value of epsilon in MAPE and ended up having to implement my own

If this is not intentional, I would like to work on adding it

**Comment by SkafteNicki** [comment]
Hi @Abdelrhman-Hosny thanks for raising this issue.
Feel free to send a PR which adds `epsilon` as an argument.

**Comment by Abdelrhman-Hosny** [comment]
I've added the argument but I'm unsure how would I write tests for it as the epsilon argument doesn't exist in `sklearn.metrics` function which is how current tests are done.

should I create a new file called `test_mape.py` and do my own tests instead ?

**Comment by SkafteNicki** [comment]
No, it would still keep it all in the `test_mean_errors.py`

**Comment by SkafteNicki** [comment]
@Abdelrhman-Hosny what are the actual use cases for changing epsilon?

**Comment by Abdelrhman-Hosny** [comment]
I was working on some data that ranges from 0 -> 40k and any small overprediction when the label is 0 predicting 1 which is mot that far off ruins the whole metric.

So by setting epsilon to a higher number, that would make the results better.

**Comment by SkafteNicki** [comment]
What do you mean by label is 0 and 1 in this case?
MAPE is an regression metric so I do not see the need for labels when using this metric.

**Comment by Abdelrhman-Hosny** [comment]
I was talking about 1 as a value not a label, a prediction of 1 when the label is zero results in a very large number (1 -0)/1e-06 which results in the mean not representing the actual error correctly just because of mispredicting the value of 0 as 1.

**Comment by SkafteNicki** [comment]
Actually, when I look at the description Sklearn have:
![image](https://user-images.githubusercontent.com/24896311/191937862-dc3fa5e6-927e-46bd-8f50-fd190ef62135.png)
taken from here: 
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_absolute_percentage_error.html
it very much sound like what you are describing is a feature/shortcoming of this particular metric, and have nothing to do with the epsilon parameter. Namely that if values are close to 0 it will heavy penalize predictions that are far away.

**Comment by Borda** [comment]
@SkafteNicki shall we add this epsilon to the base `Metric` class? :rabbit:

**Comment by SkafteNicki** [comment]
> @SkafteNicki shall we add this epsilon to the base `Metric` class? 🐰

No, not if this would be the only metric. Also, as pointed out in my last comment, this epsilon parameters are only meant for numerical stability and should therefore not be exposed to the user. Closing issue.
